### DISCUSSION TITLE
JSON scheme for mqtt message

### DISCUSSION THREAD

**Discussion by milo1000** [Koenkk/zigbee2mqtt#5724] [General]
I wanted to interface my java app with zigbee2mqtt. To do that I was **looking for JSON scheme**, so I can use some tool to create POJO and use deserializer, like GSON. Unfortunately I couldn't find one (scheme, not deserializer). **Is there such a thing?**
 Eventually, I've used [MQTT topics and message](https://www.zigbee2mqtt.io/information/mqtt_topics_and_message_structure.html) documentation, but I am not sure if it covers whole scheme. Besides, there are some obvious mistakes, documentation says:
`"exposes":{"type":"light","features":["state","brightness","color_xy"]}`
while in fact, reported message contains array:
`"exposes":[{"type":"light","features":["state","brightness","color_xy"]}]`

**Comment by Koenkk** [comment]
The JSON scheme can indeed be reconstructed from the exposes. This is also how the frontend does it.

I've fixed the typo in the docs.

**Reply by milo1000** [reply]
Thanks for info!
